### ISSUE TITLE
Heading text incorrect when there is link in heading

### ISSUE THREAD

**Issue by Cantej** [Aidurber/obsidian-plugin-dynamic-toc#69]
I have a heading with a link and some text before and after it, as shown below. 
<img width="565" alt="image" src="https://user-images.githubusercontent.com/35187140/177259340-7d470ebd-6892-47b7-902d-eb254e9bb41c.png">

### Expected Table of Contents
<img width="481" alt="image" src="https://user-images.githubusercontent.com/35187140/177259427-5459d084-2ccc-4aff-8db2-d46ff91b071e.png">

### What is happening now
<img width="509" alt="image" src="https://user-images.githubusercontent.com/35187140/177259499-f68a125a-d251-4df5-b17f-e70a0c33559e.png">

So to sum it up, text before and after a link in a heading do not render in the table of contents for some reason. 

On a side note, thank you so much for this plugin, it is SOOOOO good and has made my obsidian experience much better. Thank you for all the time and effort you put into this plugin
